src/device/wg_controller.rs

```diff
@@ -521,35 +523,41 @@ async fn build_requested_peers_list<
         requested_peers.insert(dns_peer_public_key, requested_peer);
     }
 
-    if let Some(wg_stun_server) = &requested_state.wg_stun_server {
-        let public_key = wg_stun_server.public_key;
-        let endpoint = SocketAddr::new(IpAddr::V4(wg_stun_server.ipv4), wg_stun_server.stun_port);
-        telio_log_debug!("Configuring wg-stun peer: {}, at {}", public_key, endpoint);
-        let persistent_keepalive_interval = requested_state.keepalive_periods.stun;
-        let allowed_ips = if features.ipv6 {
-            vec![
-                IpNetwork::V4("100.64.0.4/32".parse()?),
-                IpNetwork::V6("fd74:656c:696f::4/128".parse()?),
-            ]
-        } else {
-            vec![IpNetwork::V4("100.64.0.4/32".parse()?)]
-        };
-        let ip_addresses = allowed_ips.iter().copied().map(|ip| ip.ip()).collect();
+    if let (Some(wg_stun_server), Some(stun)) = (&requested_state.wg_stun_server, stun_ep_provider)
+    {
+        if !stun.is_paused().await {
+            let public_key = wg_stun_server.public_key;
+            let endpoint =
+                SocketAddr::new(IpAddr::V4(wg_stun_server.ipv4), wg_stun_server.stun_port);
+            telio_log_debug!("Configuring wg-stun peer: {}, at {}", public_key, endpoint);
+            let persistent_keepalive_interval = requested_state.keepalive_periods.stun;
+            let allowed_ips = if features.ipv6 {
+                vec![
+                    IpNetwork::V4("100.64.0.4/32".parse()?),
+                    IpNetwork::V6("fd74:656c:696f::4/128".parse()?),
+                ]
+            } else {
+                vec![IpNetwork::V4("100.64.0.4/32".parse()?)]
+            };
+            let ip_addresses = allowed_ips.iter().copied().map(|ip| ip.ip()).collect();
 
-        requested_peers.insert(
-            public_key,
-            RequestedPeer {
-                peer: telio_wg::uapi::Peer {
-                    public_key,
-                    endpoint: Some(endpoint),
-                    ip_addresses,
-                    persistent_keepalive_interval,
-                    allowed_ips,
-                    ..Default::default()
+            requested_peers.insert(
```

Comment:
Note that this logic will remove WG-STUN peer completely if STUN EP is paused [here](https://github.com/NordSecurity/libtelio/blob/LLT-4935_do_not_rotate_stun_peers/src/device/wg_controller.rs#L190) as opposed to "leaving it untouched", like PR description states, therefore I am not sure if this is intentional.

On the other hand, removing WG-STUN peer, when no STUN requests are anticipated (e.g. EP provider is paused) is probably correct thing to do. I just wonder, about possible side-effects.

Reply:
This was a goal, to remove the wg-stun from peers list, to suppress sending all those keepalives.

Reply:
Updated description